src/http/h1/decode.rs

```diff
@@ -1,39 +1,14 @@
 use std::cmp;
 use std::io::{self, Read};
 
-use self::Kind::{Length, Chunked, Eof};
+use self::Decoder::{Length, Chunked, Eof};
 
 /// Decoders to handle different Transfer-Encodings.
 ///
 /// If a message body does not include a Transfer-Encoding, it *should*
 /// include a Content-Length header.
 #[derive(Debug, Clone)]
-pub struct Decoder {
-    kind: Kind,
```

Comment:
This was done to keep the internals private. As a public enum, the internals are exposed and it'd be a breaking change later on to change them, since someone could have depended on the exposed internals.